### PR TITLE
docs: remove all references to the deprecated OSS version

### PR DESCRIPTION
**Pull request by coury-clark**
Scrubbing references to the deprecated OSS version from the documentation. There were a few places it seemed like leaving the reference made sense, so I annotated with a note it is deprecated.

## Test plan

I started Sourcegraph using sg, that seemed useful to test.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
cc @ryphil in case you have any objections

**Review by sashaostrikov — APPROVED** [review]

File: doc/dev/background-information/insights/frontend.md [hunk 1]
```diff
@@ -259,10 +259,10 @@ This page renders an insight grid component with all insights that you have in y
 this is kind of analog of the All insights dashboard on the dashboard page.
 
 But this page uses a slightly different approach how to load insights data. The directory and search pages render `ExtensionViewsSection`
-component. This component is different from the OSS to Enterprise version. But in both cases, this component is responsible for loading
+component. This component is different from the OSS to Enterprise version (note, the OSS version is deprecated as of 5.1). But in both cases, this component is responsible for loading
 extension and insight-like views and render them into the grid layout views component as it's shown in the picture above.
 
-- In OSS version it renders only extension views [source](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/insights/sections/ExtensionViewsSection.tsx)
+- In OSS version (deprecated in 5.1)) it renders only extension views [source](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/insights/sections/ExtensionViewsSection.tsx)
```
Comment (unknwon) [inline doc/dev/background-information/insights/frontend.md:265]:
```suggestion
- In OSS version (deprecated in 5.1) it renders only extension views [source](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/insights/sections/ExtensionViewsSection.tsx)
```

**Review by unknwon — COMMENTED** [review]
Quick general comment, shouldn't we say "removed" instead of "deprecated"? Because the latter implies one can still choose if they insist, which isn't the case for 5.1?

**Comment by coury-clark** [comment]
@unknwon That's a fair suggestion, I'll make that change

**Pull request merged**